lib/Sema/CSDiagnostics.cpp

```diff
@@ -5337,10 +5337,16 @@ bool ExtraneousReturnFailure::diagnoseAsError() {
     if (FD->getResultTypeRepr() == nullptr &&
         FD->getParameters()->getStartLoc().isValid() &&
         !FD->getBaseIdentifier().empty()) {
-      auto fixItLoc = Lexer::getLocForEndOfToken(
-          getASTContext().SourceMgr, FD->getParameters()->getEndLoc());
+      SourceLoc loc;
+      if (auto *whereClause = FD->getTrailingWhereClause()) {
+        loc = whereClause->getWhereLoc();
+      } else {
+        loc = FD->getBody()->getLBraceLoc();
```

Comment:
I don't know if that matters that much at all but if `{` is on a newline wouldn't this result in result type on that line instead of where the `func` is?

Reply:
Ah good point, I forgot people write code like that 🙂 I guess that is weird if it's the style where just the opening brace is on the next line

Reply:
Yeah, maybe it's not worth it to attempt and put it on the same line due to the fact that there are multiple possibilities of how throws, async and where could be arranged (even though it might be an error)...

Reply:
I changed this to move the fix-it location to after the effects specifiers if they exist, and only to the end of `throws` if its location is after `async`.

Reply:
Sounds great, thank you!